### ISSUE TITLE
link from twitter announcement of new package results in package not found

### ISSUE THREAD

**Issue by heckj** [SwiftPackageIndex/SwiftPackageIndex-Server#2217]
**Please describe the bug**

I follow the twitter feed for SPI on new packages. The sentry update was announced at https://twitter.com/SwiftPkgUpdates/status/1615057390272974873, and included a link to the site and page:
https://swiftpackageindex.com/getsentry/sentry-cocoa#releases

However, that link reports "Package not found".

The same links are provided on the main page currently for "recent releases". I tried a search for Sentry to see if was in there under a slightly different name, but to no avail...

**Explain the steps needed to reproduce the bug**

1. Navigate to to '[twitter announcement](https://twitter.com/SwiftPkgUpdates/status/1615057390272974873)'
2. Click on link to package
3. See an error: Package Not Found

**What was the expected behaviour?**

Link to a package index entry and details about Sentry

**What web browser were you using? (if relevant)**

If you're reporting a bug with the front end, please include your browser information below:

- Platform: iOS/iPad - NetNewsWire, Twitter, and direct Safari
- Browser: Safari
- Version: iOS 16.2

**Comment by finestructure** [comment]
Thanks for the report, Joe! It looks bizarre but I think I know what's happening. The package doesn't have a "default branch" version, i.e. there's no version for the branch that's designated the default branch in Github. We use info from this version for various things on the package page and while I haven't confirmed it, IIRC it's one of the only remaining ways we'll throw up a 404 because we can't resolve package information these days. (We used to do this for other reasons in the past but have made this more resilient.)

There's no easy way to tell looking at the project's Github but I believe this is happening because they changed their default branch. If so, this should fix itself the next time we index the package's versions.

The problem is that syncing from Github and syncing from our git checkouts isn't - um - synchronised =) Meaning we see the new default branch version in Github and only later crawl the checkouts again to match the versions.

Depending on where we actually throw up the 404 we could probably be more lenient here and look at the latest release version in this (rare) case to avoid the 404, or at least this particular one. Releases don't usually move, which would make this more stable.

There'd still be a gap if a package without releases changes its default branch but that's even rarer and probably something we could live with for packages that young.

Let's keep this open to a) see if it fixes itself and b) have a look if we can avoid the 404 by looking at the release version.

**Comment by heckj** [comment]
Ah, yeah - I found the commit, and it looks like I just happened across it at the wrong time. They updated that branch 5 hours ago or such: https://github.com/getsentry/sentry-cocoa/commit/feba2be7751fb57d122800cd1964eeef43aaac61

**Comment by finestructure** [comment]
Ah, well spotted, thanks!

**Comment by finestructure** [comment]
Ok, there’s more at play here, because this should have corrected itself by now (and it hasn’t).

**Comment by finestructure** [comment]
Ah, the problem is of course that `main` also had enough activity at the same time to keep it in the 24h commit coalescing window, that's why analysis isn't picking it up yet:

```
[ INFO ] throttled 1 incoming revisions [component: analyze]
```

**Comment by finestructure** [comment]
It gets weirder though. We've also got 8.0.0 as a `branch` in our db. I believe this could only stem from `8.0.0` being the default branch at some point while we indexed the package. The GH action yml file diff Joe linked to leads me to believe it might have been, as part of the transition (`branch: 8.0.0`).

<img width="668" alt="CleanShot 2023-01-17 at 06 16 24@2x" src="https://user-images.githubusercontent.com/65520/212816046-17770b5b-f96d-45b8-b432-3497c34130d2.png">

I think we could streamline this a bit if we didn't throttle renamed default branches (should be feasible). However, for this particular case I actually want to confirm that it resolves on its own by this afternoon to make sure we deal with this weird "tag as a branch" state correctly.

(I do have a db snapshot for further testing with the weird state in case it doesn't.)
